projects/media-viewer/src/lib/shared/toolbar.events.service.ts

```diff
@@ -0,0 +1,36 @@
+import {Injectable} from '@angular/core';
+import {BehaviorSubject} from 'rxjs';
+
+// Toolbar Custom-Event Types
+export type HighlightMode = boolean;
+export type DrawMode = boolean;
+
+@Injectable({providedIn: 'root'})
+export class ToolbarEventsService {
+  // Register Observable Subject Events relevant to the Toolbar
+  highlightMode = new BehaviorSubject<HighlightMode>(false);
+  drawMode = new BehaviorSubject<DrawMode>(false);
+  constructor() {}
+
+  // Function to inform Observers that highlightMode has been enabled
+  public toggleHighlightMode(): void {
+    // Highlight and Draw states are mutually exclusive
+    if (this.highlightMode.getValue() === false) {
```

Comment:
maybe this could be simplified by emitting the negative of the current value
this.drawMode.next(!this.highlightMode.getValue());
this.highlightMode.next(this.highlightMode.getValue());

Reply:
the issue with the above is the mutually exclusive behaviour. Its not a rocker switch, but instead there are multiple states. On / Off, and if you click on the other ones, they turn the respective other toggle on or off. Using negation on the current value doesn’t help to understand that logic.

Reply:
yes, it's less obvious, but the second line gives you a hint that the values are mutually exclusive

Reply:
not really no..
So scenario 1: 
They act as a toggle. Your suggested code works in that instance.
Scenario 2: 
Your code falls over as the draw mode will always be the opposite of the HighlightButton state, it doesn’t account for the instances where you want both off.

I can’t think of another way of achieving this bar a switch statement, so going to keep as is, as this breaks my tests otherwise